self_enrollment_tool/utils.py

```diff
@@ -0,0 +1,76 @@
+import logging
+
+from canvas_sdk import RequestContext
+from canvas_sdk.exceptions import CanvasAPIError
+from canvas_sdk.methods.external_tools import (create_external_tool_courses,
+                                               list_external_tools_courses)
+from canvas_sdk.utils import get_all_list_data
+from django.conf import settings
+from icommons_common.canvas_api.helpers import \
+    roles as canvas_api_helpers_roles
+from icommons_common.models import UserRole
+
+logger = logging.getLogger(__name__)
+
+SDK_SETTINGS = settings.CANVAS_SDK_SETTINGS
+# make sure the session_inactivity_expiration_time_secs key isn't in the settings dict
+SDK_SETTINGS.pop('session_inactivity_expiration_time_secs', None)
+SDK_CONTEXT = RequestContext(**SDK_SETTINGS)
+
+
+def get_canvas_roles():
+    logger.debug(f"called _get_canvas_roles()")
+    roles = []
+    try:
+        canvas_roles = canvas_api_helpers_roles.get_roles_for_account_id('self')
+        user_roles = UserRole.objects.filter(pk__in=settings.SELF_ENROLLMENT_TOOL_ROLES_LIST)
+        for role in user_roles:
+            label = canvas_roles[role.canvas_role_id]['label']
+            roles.append({'roleId': role.role_id, 'roleName': label})
+    except CanvasAPIError:
+        logger.exception("Failed to retrieve roles for the root account from Canvas API")
+        return None
+    except Exception:
+        logger.exception("Unhandled exception in _get_canvas_roles; aborting.")
+        return None
+
+    roles.sort(key=lambda x: x['roleName'])
+    logger.debug(f"returning roles: {roles}")
+    return roles
+
+
+def install_unenrollment_tool(course_instance_id, client_id):
+    # make sure that the unenrollment tool is installed
+
+
+    if not client_id:
+        raise Exception('Could not install the self-unenrollment tool into the Canvas course site: the tool client_id is missing.')
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.



Similar issue at line number 76.